pwiz_tools/Skyline/Model/Koina/Models/KoinaModel.cs

```diff
@@ -705,26 +705,79 @@ public static double CalculateSpectrumDotpMzMatch(LibraryRankedSpectrumInfo spec
                 new Statistics(intensities2All));
         }
 
-        /*public static double CalculateSpectrumDotpIonMatch(LibraryRankedSpectrumInfo spectrum1,
-            LibraryRankedSpectrumInfo spectrum2)
+        /// <summary>
+        /// Calculates the normalized contrast angle of the square rooted
+        /// intensities of the two given spectra using all peaks and possibly ignoring
+        /// mismatched peaks
+        /// </summary>
+        /// <param name="spectrum1">First spectrum</param>
+        /// <param name="spectrum2">Second spectrum</param>
+        /// <param name="mzTolerance">Tolerance for considering two mzs the same</param>
+        /// <param name="ignoreMismatches1">if true, unmatched peaks from the first spectrum are ignored</param>
+        /// <param name="ignoreMismatches2">if true, unmatched peaks from the second spectrum are ignored</param>
+        public static double CalculateSpectrumDotpMzFull(LibraryRankedSpectrumInfo spectrum1,

```

Comment:
I think `CalculateSpectrumDotpMzMatch` should be changed to call `CalculateSpectrumDotpMzFull` passing in `false` for `ignoreMismatches1` and `ignoreMismatches2` so that there is less duplicated code.

I think the implementation in "CalculateSpectrumDotpMzMatch" is a lot cleaner in terms of iterating over the "RankedMI" lists in the two spectra. I think it would be better if "CalculateSpectrumDotpMzFull" looked more like the way "CalculateSpectrumDotpMzMatch" looks like now.